### PR TITLE
Implement studies_order_by query params in studies list

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

https://github.com/optuna/optuna-dashboard/discussions/332#discussioncomment-6515533

## What does this implement/fix? Explain your changes.

Implement studies_order_by query params in studies list

pr

### CODE REVIEW

File: optuna_dashboard/ts/components/StudyList.tsx [hunk 1]
```diff
@@ -56,17 +57,42 @@ export const StudyList: FC<{
...
+  useEffect(() => {
+    const handlePopState = () => {
+      const params = new URLSearchParams(window.location.search)
+      setSortBy(params.get("studies_order_by") || "asc")
+    }
+    window.addEventListener("popstate", handlePopState)
+    return () => {
+      window.removeEventListener("popstate", handlePopState)
+    }
+  }, [])
```
Comment (c-bata) [inline optuna_dashboard/ts/components/StudyList.tsx:87]: Could you please explain the role of this logic? It looks that works as expected without this.

**Review by c-bata — COMMENTED** [review]
@keisuke-umezawa Thank you for your pull request. I left some questions.

File: optuna_dashboard/ts/components/StudyList.tsx [hunk 2]
```diff
@@ -56,17 +57,42 @@ export const StudyList: FC<{
     useDeleteStudyDialog()
   const [openRenameStudyDialog, renderRenameStudyDialog] =
     useRenameStudyDialog(studies)
-  const [sortBy, setSortBy] = useState<"id-asc" | "id-desc">("id-asc")
+
+  const navigate = useNavigate()
+  const location = useLocation()
+  const queryParams = new URLSearchParams(location.search)
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply optuna_dashboard/ts/components/StudyList.tsx]: Fixed.

File: optuna_dashboard/ts/components/StudyList.tsx (hunk 1 above)
Comment (keisuke-umezawa) [reply optuna_dashboard/ts/components/StudyList.tsx]: You are right. While doing trial and error, I added this unnecessary code. Thank you for your review. I removed them.

**Comment by keisuke-umezawa** [comment]
@c-bata Could you review it again?

**Review by c-bata — APPROVED** [review]
LGTM!

**Pull request merged**